app/src/main/java/com/spascoding/taskycourse/feature_register_screen/presentation/register/RegisterScreen.kt

```diff
@@ -32,14 +33,44 @@ import com.spascoding.taskycourse.core.constants.Padding
 import com.spascoding.taskycourse.core.constants.RoundCorner
 import com.spascoding.taskycourse.feature_register_screen.presentation.components.CustomOutlinedTextField
 import com.spascoding.taskycourse.feature_register_screen.presentation.components.PasswordOutlinedTextField
+import com.spascoding.taskycourse.feature_register_screen.presentation.util.AuthPattern
 import com.spascoding.taskycourse.navigation.Navigation
+import com.spascoding.taskycourse.ui.theme.TaskyCourseTheme
 
 @Composable
-fun RegisterScreen(
+fun RegisterScreenRoot(
     navController: NavController,
     viewModel: RegisterViewModel = hiltViewModel()
 ) {
     val state by viewModel.state.collectAsStateWithLifecycle()
+
+    val onNavigationEvent: (RegisterEvent) -> Unit = { event ->
+        when (event) {
+            is RegisterEvent.BackAction -> {
+                navController.popBackStack()
```

Comment:
For user-initiated back stack popping, I'd prefer `navigateUp`, since that is aware of your back stack and will only navigate to the previous screen (and not generally just pop one from the back stack whatever it is).
When this is done by `popBackStack`, clicking a button and there's a transition animation or so, it can happen that the user could press the button multiple times when tapped fast. That would clear more of the back stack as you'd want here